dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go

```diff
@@ -77,6 +77,13 @@ func RunGenerateCRD(ctx context.Context, o *GenerateCRDOptions) error {
 	if o.GenerateOptions.ProtoSourcePath == "" {
 		return fmt.Errorf("`proto-source-path` is required")
 	}
+	if o.ResourceProtoName == "" {
+		return fmt.Errorf("`--proto-resource` is required")
+	}
+	if o.ResourceKindName == "" {
+		return fmt.Errorf("`--kind` is required")
+	}
+	o.ResourceProtoName = capitalizeFirstRune(o.ResourceProtoName)
```

Comment:
What happens if neither is provided previously? IIRC, if --kind and --proto-resource is not given, this tool generates everything except the "<kind>_types.go" file. Maybe we'd better still keep this functionality and only require both if either one is given.